Makefile

```diff
@@ -0,0 +1,108 @@
+# Makefile for easier installation and cleanup.
+#
+# Uses self-documenting macros from here:
+# http://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
+
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+MAKEFLAGS += --warn-undefined-variables --no-builtin-rules
+
+PACKAGE=readabilipy
+DOC_DIR=./docs
+VENV_DIR=/tmp/rdpy_venv
+TEST_DIR=./tests
+
+.PHONY: help
+
+.DEFAULT_GOAL := help
+
+help:
+	@grep -E '^[0-9a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) |\
+		 awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-15s\033[0m\
+		 %s\n", $$1, $$2}'
+
+################
+# Installation #
+################
+
+.PHONY: install
+
+install: ## Install for the current user using the default python command
+	python setup.py build_ext --inplace
+	python setup.py install --user
+
+
+################
+# Distribution #
+################
+
+.PHONY: release dist
+
+release: ## Make a release
+	python make_release.py
+
+dist: ## Make Python source distribution
+	python setup.py sdist bdist_wheel
+
+
+###########
+# Testing #
+###########
+
+.PHONY: test
+
+test: venv ## Run unit tests
+	source $(VENV_DIR)/bin/activate && cd $(TEST_DIR) && \
+		python -m pytest -v . --cov readabilipy \
+		--cov-report term-missing --benchmark-disable
+	source $(VENV_DIR)/bin/activate && pyflakes *.py readabilipy tests
+	source $(VENV_DIR)/bin/activate && pycodestyle --statistics \
+		--ignore=E501 --count *.py readabilipy tests
+	source $(VENV_DIR)/bin/activate && pylint readabilipy
+
+#################
+# Documentation #
+#################
+
+.PHONY: doc docs
+
+docs: doc
+doc: install ## Build documentation with Sphinx
+	exit; # not implemented
+	source $(VENV_DIR)/bin/activate && m2r README.md && mv README.rst $(DOC_DIR)
+	source $(VENV_DIR)/bin/activate && m2r CHANGELOG.md && mv CHANGELOG.rst $(DOC_DIR)
+	cd $(DOC_DIR) && \
+		rm source/* && \
+		source $(VENV_DIR)/bin/activate && \
+		sphinx-apidoc -H 'Paper2Remarkable API Documentation' -o source ../$(PACKAGE) && \
+		touch source/AUTOGENERATED
+	$(MAKE) -C $(DOC_DIR) html
```

Comment:
Are these docs currently useful? From what I remember, we didn't stick to Doxygen/Sphinx documentation standards. Should we consider a separate PR to tidy up the docs?

Reply:
Yeah, I think what happened here is I used a "standard" Makefile I use for Python packaging but disabled this target for now by adding the ``exit`` statement. We can remove the target altogether if we don't have docs.

Reply:
I'd say yes to a PR to set up the docs, btw.

Reply:
Docs issue at #88